File: 700/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def searchBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        # base case
        if root is None or root.val == val:
            return root
        # check if val < less than curr val
        if val < root.val:
        # go to left child
            return self.searchBST(root.left, val)
        else:
        # go to right child
            return self.searchBST(root.right, val)

File: 700/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
    // base case
    if (root == null || root.val == val)
        return root
    // check if val < current val
    if (val < root.val)
    // go to left child
        return searchBST(root.left, val)
    // go to right child
    else
        return searchBST(root.right, val)
};